### PR TITLE
NPE empty attributes

### LINKED ISSUE #9160: NullPointerException: Attempt to invoke virtual method 'com.google.gson.JsonArray com.google.gson.JsonElement.getAsJson...
Sentry Issue: [WOOCOMMERCE-ANDROID-6MK](https://a8c.sentry.io/issues/4126355699/?referrer=github_integration)

```
NullPointerException: Attempt to invoke virtual method 'com.google.gson.JsonArray com.google.gson.JsonElement.getAsJsonArray()' on a null object reference
    at com.woocommerce.android.model.ProductKt.toAppModel(Product.kt:539)
    at com.woocommerce.android.model.ProductKt.toAppModel(Product.kt:539)
    at com.woocommerce.android.model.ProductKt.toAppModel(Product.kt:539)
    at com.woocommerce.android.ui.products.ProductListRepository.getProductList(ProductListRepository.kt:175)
    at com.woocommerce.android.ui.products.ProductListRepository.getProductList$default(ProductListRepository.kt:163)
...
(41 additional frame(s) were not displayed)
```

### LINKED ISSUE #9162: NullPointerException: Attempt to invoke virtual method 'com.google.gson.JsonArray com.google.gson.JsonElement.getAsJson...
Sentry Issue: [WOOCOMMERCE-ANDROID-6W9](https://a8c.sentry.io/issues/4194779192/?referrer=github_integration)

```
NullPointerException: Attempt to invoke virtual method 'com.google.gson.JsonArray com.google.gson.JsonElement.getAsJsonArray()' on a null object reference
    at com.woocommerce.android.model.ProductKt.toAppModel(Product.kt:535)
    at com.woocommerce.android.model.ProductKt.toAppModel(Product.kt:535)
    at com.woocommerce.android.model.ProductKt.toAppModel(Product.kt:535)
    at com.woocommerce.android.ui.products.ProductListRepository.getProductList(ProductListRepository.kt:175)
    at com.woocommerce.android.ui.products.ProductListRepository.getProductList$default(ProductListRepository.kt:163)
...
(59 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by atorresveiga**
Closes: https://github.com/woocommerce/woocommerce-android/issues/9160
Closes: https://github.com/woocommerce/woocommerce-android/issues/9162

### Description
This PR solves the NPE crash in the app when receiving unexpected values in the attributes field. We previously caught only JsonParseException parsing exceptions, and now we are swallowing all types of exceptions. Now, we return an empty list whenever we fail to parse the product attributes (the same behavior we had before but only for JsonParseException).

Here p1686673876637129-slack-CGPNUU63E, we discussed about the WCProductModel refactor to be able to unit tests the attributes deserialization.

### Testing
Check that exceptions deserializing product's attributes don't crash the app

### CODE REVIEW

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/model/WCProductModel.kt [hunk 1]
```diff
@@ -317,6 +317,11 @@ data class WCProductModel(@PrimaryKey @Column private var id: Int = 0) : Identif
         return null
     }
 
+    /**
+     * Returns the list of products attributes. The function returns an empty list
+     * when the attributes json deserialization fails.
+     */
+    @Suppress("SwallowedException", "TooGenericExceptionCaught")
```
Comment (ThomazFB) [inline plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/model/WCProductModel.kt]: np: I think you don't need the "SwallowedException" supression here.

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/model/WCProductModel.kt [hunk 2]
```diff
@@ -347,7 +352,7 @@ data class WCProductModel(@PrimaryKey @Column private var id: Int = 0) : Identif
                     )
                 }
             }
-        } catch (e: JsonParseException) {
+        } catch (e: Exception) {
```
Comment (ThomazFB) [inline plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/model/WCProductModel.kt]: There's a way to refactor this while keeping the generic catch, avoiding the `TooGenericExceptionCaught` supression, and improve the overall performance of the operation while keeping it idiomatic:

```
        return kotlin.runCatching {
            Gson().fromJson(attributes, JsonElement::class.java)
                .asJsonArray.asSequence()
                .map { it.asJsonObject }
                .map { json ->
                    ProductAttribute(
                        id = json.getLong("id"),
                        name = json.getString("name") ?: "",
                        variation = json.getBoolean("variation", true),
                        visible = json.getBoolean("visible", true),
                        options = getAttributeOptions(json.getAsJsonArray("options"))
                    )
                }.toList()
        }.fold(
            onSuccess = { it },
            onFailure = { emptyList() }
        )
```

By using the `runCatching`, we make sure that every single `Throwable` coming from that block will trigger the `onFailure`, allowing us to remove all suppression from the top of the function and making it clear what to do on each scenario through the result folding. 

WDYT? Makes sense?

File: plugins/woocommerce/src/test/java/org/wordpress/android/fluxc/model/WCProductModelTest.kt [hunk 3]
```diff
@@ -0,0 +1,42 @@
+package org.wordpress.android.fluxc.model
+
+import com.google.gson.JsonArray
+import com.google.gson.JsonObject
+import org.assertj.core.api.Assertions
+import org.junit.Test
+
+class WCProductModelTest {
```
Comment (ThomazFB) [inline plugins/woocommerce/src/test/java/org/wordpress/android/fluxc/model/WCProductModelTest.kt:8]: Great work adding unit tests!

**Review by ThomazFB — APPROVED** [review]
Tests are good and seem to work just fine! I've left some suggestions regarding how to remove the lint suppressions and keep the same strategy, but nothing blocking!

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/model/WCProductModel.kt (hunk 2 above)
Comment (atorresveiga) [reply plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/model/WCProductModel.kt]: TIL about `kotlin.runCatching`, great recommendation @ThomazFB 👍 
Fixed here [3a00135](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2752/commits/3a0013550c87cec2fc699549d6eb15511770df9a)

**Comment by ThomazFB** [comment]
Thank you for moving forward with my suggestion, @atorresveiga! Looks good!

**Pull request merged**